### ISSUE TITLE
Move symbol refactoring not updating the import when symbol is referenced inside a function

### ISSUE THREAD

**Issue by luabud** [microsoft/pylance-release#4234]
## Environment data

-   Language Server version: 2023.4.11
-   OS and version: Windows 11
-   Python version (& distribution if applicable, e.g. Anaconda): 3.11.3, venv 

## Code Snippet

```python
def myfunc():
    x = 10
    y = 20
    z = x+y
    print(z)
```

## Repro Steps

1. Select lines 2-4 (from x = 10 to z = x+y)
2. Run "Extract Method" and call it e.g. mysum
3. This will generate the following code:
```
def myfunc():
    z = mysum()
    print(z)

def mysum():
    x = 10
    y = 20
    z = x+y
    return z
```

4. Click on mysum and run "Move symbol to new file"

## Expected behavior
The symbol is moved and `from mysum import mysum` is added to the top of the file 

## Actual behavior

he symbol is moved, but the reference on the second line (z = mysum()) is not recognized because the import statement isn't added to the file

## Logs

```
[Info  - 1:24:37 PM] (23212) Pylance language server 2023.4.11 (pyright 64a33975) starting
[Info  - 1:24:37 PM] (23212) Server root directory: c:\REDACTED\.vscode-insiders\extensions\ms-python.vscode-pylance-2023.4.11\dist
[Info  - 1:24:37 PM] (23212) Starting service instance "wordle"
[Info  - 1:24:37 PM] (23212) Auto-indent enabled
[Info  - 1:24:37 PM] (23212) Setting pythonPath for service "wordle": "c:\REDACTED\Apps\wordle\.venv\Scripts\python.exe"
[Warn  - 1:24:37 PM] (23212) stubPath c:\REDACTED\Apps\wordle\typings is not a valid directory.
[Info  - 1:24:37 PM] (23212) Assuming Python version 3.11
[Info  - 1:24:38 PM] (23212) Found 2 source files
[Info  - 1:24:38 PM] (23212) Background analysis(1) root directory: c:\REDACTED\.vscode-insiders\extensions\ms-python.vscode-pylance-2023.4.11\dist
[Info  - 1:24:38 PM] (23212) Background analysis(1) started
[Info  - 1:24:38 PM] (23212) Indexer background runner(2) root directory: c:\REDACTED\.vscode-insiders\extensions\ms-python.vscode-pylance-2023.4.11\dist (index)
[Info  - 1:24:38 PM] (23212) Indexing(2) started
[Info  - 1:24:40 PM] (23212) scanned(2) 14 files over 1 exec env
[Info  - 1:24:40 PM] (23212) indexed(2) 4 files over 1 exec env
[Info  - 1:24:40 PM] (23212) Indexing finished(2).
[Info  - 1:25:14 PM] (23212) Found 3 source files
[Info  - 1:25:52 PM] (23212) Found 2 source files
[Info  - 1:44:18 PM] (23212) Found 3 source files
[Info  - 1:44:21 PM] (23212) Found 2 source files
[Info  - 1:46:33 PM] (23212) Found 3 source files
[Info  - 1:46:35 PM] (23212) Found 2 source files
[Info  - 1:47:48 PM] (23212) Found 3 source files
[Info  - 1:48:41 PM] (23212) Found 4 source files
[Info  - 1:48:48 PM] (23212) Found 3 source files
[Info  - 1:49:07 PM] (23212) Found 4 source files
[Info  - 1:49:09 PM] (23212) Found 3 source files
[Info  - 1:49:26 PM] (23212) Found 4 source files
[Info  - 1:49:27 PM] (23212) Found 3 source files
[Info  - 1:54:12 PM] (23212) Found 4 source files
[Info  - 1:54:36 PM] (23212) Setting pythonPath for service "wordle": "c:\REDACTED\Apps\wordle\.venv\Scripts\python.exe"
[Warn  - 1:54:36 PM] (23212) stubPath c:\REDACTED\Apps\wordle\typings is not a valid directory.
[Info  - 1:54:36 PM] (23212) Assuming Python version 3.11
[Info  - 1:54:36 PM] (23212) Found 4 source files
[Info  - 1:54:37 PM] (23212) Indexer background runner(3) root directory: c:\REDACTED\.vscode-insiders\extensions\ms-python.vscode-pylance-2023.4.11\dist (index)
[Info  - 1:54:37 PM] (23212) Indexing(3) started
[Info  - 1:54:37 PM] (23212) scanned(3) 14 files over 1 exec env
[Info  - 1:54:37 PM] (23212) indexed(3) 4 files over 1 exec env
[Info  - 1:54:37 PM] (23212) Indexing finished(3).
[Info  - 1:54:41 PM] (23212) Found 3 source files
[Info  - 1:55:16 PM] (23212) Setting pythonPath for service "wordle": "c:\REDACTED\Apps\wordle\.venv\Scripts\python.exe"
[Warn  - 1:55:16 PM] (23212) stubPath c:\REDACTED\Apps\wordle\typings is not a valid directory.
[Info  - 1:55:16 PM] (23212) Assuming Python version 3.11
[Info  - 1:55:16 PM] (23212) Found 3 source files
[Info  - 1:55:17 PM] (23212) Indexer background runner(4) root directory: c:\REDACTED\.vscode-insiders\extensions\ms-python.vscode-pylance-2023.4.11\dist (index)
[Info  - 1:55:17 PM] (23212) Indexing(4) started
[Info  - 1:55:17 PM] (23212) scanned(4) 14 files over 1 exec env
[Info  - 1:55:17 PM] (23212) indexed(4) 4 files over 1 exec env
[Info  - 1:55:17 PM] (23212) Indexing finished(4).
[Info  - 1:55:34 PM] (23212) Found 4 source files

```

**Comment by rchiodo** [comment]
@Heejaechang are you already fixing this?

**Comment by heejaechang** [comment]
> @heejaechang are you already fixing this?

oops. sorry I didn't see this.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.4.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023441-26-April-2023-PreRelease)
